gui/wxpython/datacatalog/catalog.py

```diff
@@ -103,6 +103,7 @@ def OnAddGrassDB(self, event):
         if dlg.ShowModal() == wx.ID_OK:
             grassdatabase = dlg.GetPath()
             self.tree.InsertGrassDb(name=grassdatabase)
+            self.tree.OfferCreateLocation(grassdb=grassdatabase)
```

Comment:
The functionality should be here rather than in tree.py. When you add the database, simply just check if the directory is empty and then ask if they want to create new location.

Reply:
I like the idea of simply check if the directory is empty. But I am not sure if it should be part of catalog.py, because
1. Generally speaking no wx.MessageDialog dialogs are reated in catalog.py
2. We need to have grassdb node for calling CreateLocation function and it is a bit strange to call self.tree._model.SearchNodes
3. We could use InsertGrassDb function later in other functions e.g. in settings/GRASS working environment
I moved it to Data Catalog but would be glad to discuss it. :)

Reply:
> I like the idea of simply check if the directory is empty. But I am not sure if it should be part of catalog.py, because
> 
> 1. Generally speaking no wx.MessageDialog dialogs are created in catalog.py

that's not a good reason

> 2. We need to have grassdb node for calling CreateLocation function and it is a bit strange to call self.tree._model.SearchNodes

you don't need that, InsertGrassDb returns the node

> 3. We could use InsertGrassDb function in tree.py later in other functions e.g. in settings/GRASS working environment
> 
I am not sure I understand this one.

> I moved it to Data Catalog but would be glad to discuss it. :)

Reply:
> 
> 
> This can be further simplified, you don't need to search for the node.



> 
> 
> This can be further simplified, you don't need to search for the node.

Ahh. I have not noticed that InsertGrassDb is the function, not the procedure.. Corrected.